Build failure: ././tools/derive: getDirectoryContents:openDirStream: does not exist

We are currently seeing build failures on all CI bots:
```
././tools/derive: getDirectoryContents:openDirStream: does not exist (No such file or directory)

Error when running Shake build system:
* _build/stage1/lib/bin/unlit
* OracleQ (PackageDataFile (Context {stage = Stage1, package = Package {pkgLanguage = Haskell, pkgType = Program, pkgName = "unlit", pkgPath = "utils/unlit"}, way = v}))
* _build/stage1/utils/unlit/setup-config
* _build/stage0/bin/ghc
* OracleQ (PackageDataFile (Context {stage = Stage0, package = Package {pkgLanguage = Haskell, pkgType = Program, pkgName = "ghc-bin", pkgPath = "ghc"}, way = v}))
* _build/stage0/ghc/setup-config
* _build/stage0/lib/package.conf.d/ghc-boot-8.7.conf
* OracleQ (PackageDataFile (Context {stage = Stage0, package = Package {pkgLanguage = Haskell, pkgType = Library, pkgName = "ghc-boot", pkgPath = "libraries/ghc-boot"}, way = v}))
* _build/stage0/libraries/ghc-boot/setup-config
* _build/stage0/lib/package.conf.d/binary-0.8.5.1.conf
* OracleQ (PackageDataFile (Context {stage = Stage0, package = Package {pkgLanguage = Haskell, pkgType = Library, pkgName = "binary", pkgPath = "libraries/binary"}, way = v}))

* _build/stage0/libraries/binary/setup-config
ExitFailure 1
```

New failure:

```
$ hadrian/build.sh
Up to date
Up to date
Warning: libraries/text/text.cabal:4:1: The field "bug-reports" is specified
more than once at positions 4:1, 42:1
Warning: libraries/haskeline/haskeline.cabal:4:1: The field "category" is
specified more than once at positions 4:1, 10:1
Warning: utils/hpc/hpc-bin.cabal:11:2: Tabs used as indentation at 11:2
Warning: libraries/text/text.cabal:4:1: The field "bug-reports" is specified
more than once at positions 4:1, 42:1
| Run Ghc FindHsDependencies Stage1: libraries/ghc-compact/GHC/Compact.hs (and 1 more) => _build/stage1/libraries/ghc-compact/.dependencies.mk
| Remove file _build/stage1/libraries/ghc-compact/.dependencies.mk.bak
| Run Ghc CompileHs Stage1: libraries/ghc-compact/GHC/Compact.hs => _build/stage1/libraries/ghc-compact/build/GHC/Compact.p_o
| Run Ghc CompileHs Stage1: libraries/ghc-compact/GHC/Compact/Serialized.hs => _build/stage1/libraries/ghc-compact/build/GHC/Compact/Serialized.p_o
| Remove file _build/stage1/libraries/ghc-compact/build/libHSghc-compact-0.1.0.0_p.a
| Run Ar Pack Stage1: _build/stage1/libraries/ghc-compact/build/GHC/Compact.p_o (and 1 more) => _build/stage1/libraries/ghc-compact/build/libHSghc-compact-0.1.0.0_p.a
/usr/bin/ar: creating _build/stage1/libraries/ghc-compact/build/libHSghc-compact-0.1.0.0_p.a
/---------------------------------------------------------------------------------\
| Successfully built library 'ghc-compact' (Stage1, way p).                       |
| Library: _build/stage1/libraries/ghc-compact/build/libHSghc-compact-0.1.0.0_p.a |
| Library synopsis: In memory storage of deeply evaluated data structure.         |
\---------------------------------------------------------------------------------/
| Run Ghc CompileHs Stage1: libraries/ghc-compact/GHC/Compact.hs => _build/stage1/libraries/ghc-compact/build/GHC/Compact.o
| Run Ghc CompileHs Stage1: libraries/ghc-compact/GHC/Compact/Serialized.hs => _build/stage1/libraries/ghc-compact/build/GHC/Compact/Serialized.o
| Run Ld Stage1: _build/stage1/libraries/ghc-compact/build/GHC/Compact.o (and 1 more) => _build/stage1/libraries/ghc-compact/build/HSghc-compact-0.1.0.0.o
| Remove file _build/stage1/libraries/ghc-compact/build/libHSghc-compact-0.1.0.0.a
| Run Ar Pack Stage1: _build/stage1/libraries/ghc-compact/build/GHC/Compact.o (and 1 more) => _build/stage1/libraries/ghc-compact/build/libHSghc-compact-0.1.0.0.a
/usr/bin/ar: creating _build/stage1/libraries/ghc-compact/build/libHSghc-compact-0.1.0.0.a
/-------------------------------------------------------------------------------\
| Successfully built library 'ghc-compact' (Stage1, way v).                     |
| Library: _build/stage1/libraries/ghc-compact/build/libHSghc-compact-0.1.0.0.a |
| Library synopsis: In memory storage of deeply evaluated data structure.       |
\-------------------------------------------------------------------------------/
| Run Ghc FindHsDependencies Stage1: utils/hsc2hs/ATTParser.hs (and 9 more) => _build/stage1/utils/hsc2hs/.dependencies.mk
| Remove file _build/stage1/utils/hsc2hs/.dependencies.mk.bak
| Run Ghc CompileHs Stage1: _build/stage1/utils/hsc2hs/build/hsc2hs/autogen/Paths_hsc2hs.hs => _build/stage1/utils/hsc2hs/build/Paths_hsc2hs.o
| Run Ghc CompileHs Stage1: utils/hsc2hs/HSCParser.hs => _build/stage1/utils/hsc2hs/build/HSCParser.o
| Run Ghc CompileHs Stage1: utils/hsc2hs/Common.hs => _build/stage1/utils/hsc2hs/build/Common.o
| Run Ghc CompileHs Stage1: utils/hsc2hs/Flags.hs => _build/stage1/utils/hsc2hs/build/Flags.o
| Run Ghc CompileHs Stage1: utils/hsc2hs/C.hs => _build/stage1/utils/hsc2hs/build/C.o
| Run Ghc CompileHs Stage1: utils/hsc2hs/UtilsCodegen.hs => _build/stage1/utils/hsc2hs/build/UtilsCodegen.o
| Run Ghc CompileHs Stage1: utils/hsc2hs/ATTParser.hs => _build/stage1/utils/hsc2hs/build/ATTParser.o
| Run Ghc CompileHs Stage1: utils/hsc2hs/DirectCodegen.hs => _build/stage1/utils/hsc2hs/build/DirectCodegen.o
| Run Ghc CompileHs Stage1: utils/hsc2hs/CrossCodegen.hs => _build/stage1/utils/hsc2hs/build/CrossCodegen.o
| Run Ghc CompileHs Stage1: utils/hsc2hs/Main.hs => _build/stage1/utils/hsc2hs/build/Main.o
| Run Ghc LinkHs Stage1: _build/stage1/utils/hsc2hs/build/Main.o (and 9 more) => _build/stage1/bin/hsc2hs
_build/stage1/lib/../lib/i386-linux-ghc-8.7.20180626/rts-1.0/libHSrts-1.0.a(Adjustor.o):function createAdjustor: error: undefined reference to 'adjustorCode'
collect2: error: ld returned 1 exit status
`gcc' failed in phase `Linker'. (Exit code: 1)
shakeArgsWith   0.000s    0%                           
Function shake  0.407s    1%                           
Database read   0.349s    0%                           
With database   0.021s    0%                           
Running rules  38.289s   98%  =========================
Total          39.066s   99%                           
Error when running Shake build system:
* _build/stage1/bin/hsc2hs
user error (Development.Shake.cmd, system command failed
Command: _build/stage0/bin/ghc -Wall -hisuf hi -osuf o -hcsuf hc -static -hide-all-packages -no-user-package-db '-package-db _build/stage1/lib/package.conf.d' '-package-id base-4.12.0.0' '-package-id containers-0.6.0.1' '-package-id directory-1.3.2.3' '-package-id filepath-1.4.2' '-package-id process-1.6.3.0' -i -i_build/stage1/utils/hsc2hs/build -i_build/stage1/utils/hsc2hs/build/hsc2hs/autogen -iutils/hsc2hs/. -Iincludes -I_build/generated -I_build/stage1/utils/hsc2hs/build -I/home/sam2/src/ghc/_build/stage1/lib/i386-linux-ghc-8.7.20180626/process-1.6.3.0/include -I/home/sam2/src/ghc/_build/stage1/lib/i386-linux-ghc-8.7.20180626/unix-2.8.0.0/include -I/home/sam2/src/ghc/_build/stage1/lib/i386-linux-ghc-8.7.20180626/time-1.8.0.2/include -I/home/sam2/src/ghc/_build/stage1/lib/i386-linux-ghc-8.7.20180626/bytestring-0.10.8.2/include -I/home/sam2/src/ghc/_build/stage1/lib/i386-linux-ghc-8.7.20180626/base-4.12.0.0/include -I/home/sam2/src/ghc/_build/stage1/lib/i386-linux-ghc-8.7.20180626/integer-gmp-1.0.2.0/include -I/home/sam2/src/ghc/_build/stage1/lib/i386-linux-ghc-8.7.20180626/rts-1.0/include -I_build/generated -optc-I_build/generated -optP-include -optP_build/stage1/utils/hsc2hs/build/hsc2hs/autogen/cabal_macros.h -optc-U__i686 -optc-fno-stack-protector -optP-DIN_GHC_TREE -odir _build/stage1/utils/hsc2hs/build -hidir _build/stage1/utils/hsc2hs/build -stubdir _build/stage1/utils/hsc2hs/build -no-auto-link-packages -rtsopts -optl-lgmp -Wnoncanonical-monad-instances -optc-Werror=unused-but-set-variable -optc-Wno-error=inline _build/stage1/utils/hsc2hs/build/Main.o _build/stage1/utils/hsc2hs/build/C.o _build/stage1/utils/hsc2hs/build/Common.o _build/stage1/utils/hsc2hs/build/CrossCodegen.o _build/stage1/utils/hsc2hs/build/DirectCodegen.o _build/stage1/utils/hsc2hs/build/Flags.o _build/stage1/utils/hsc2hs/build/HSCParser.o _build/stage1/utils/hsc2hs/build/ATTParser.o _build/stage1/utils/hsc2hs/build/UtilsCodegen.o _build/stage1/utils/hsc2hs/build/Paths_hsc2hs.o -o _build/stage1/bin/hsc2hs -O2 -H32m -XHaskell2010 -ghcversion-file=/home/sam2/src/ghc/_build/generated/ghcversion.h
Exit code: 1
Stderr:
_build/stage1/lib/../lib/i386-linux-ghc-8.7.20180626/rts-1.0/libHSrts-1.0.a(Adjustor.o):function createAdjustor: error: undefined reference to 'adjustorCode'
collect2: error: ld returned 1 exit status
`gcc' failed in phase `Linker'. (Exit code: 1)
)
```

It does seem to have successfully built the stage-1 compiler, and this error looks unrelated. I think the Cabal thing is definitely fixed.

--- a/.travis.yml
+++ b/.travis.yml
@@ -104,13 +104,6 @@ install:
     - mv .git ghc/hadrian
     - cd ghc/hadrian && git reset --hard HEAD && cd ..
 
-    # We checkout a patched Cabal.
-    # See: https://github.com/snowleopard/hadrian/issues/634
-    - cd libraries/Cabal/
-    - git remote add quasicomputational https://github.com/quasicomputational/cabal.git
-    - git fetch quasicomputational && git checkout cwd-independent-check
-    - cd ../../
-
 cache:
     directories:
         - $HOME/.cabal

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -20,13 +20,6 @@ install:
     # Copy new Hadrian into ./ghc/hadrian
     - cp -r new-hadrian ghc\hadrian
 
-    # We checkout a patched Cabal.
-    # See: https://github.com/snowleopard/hadrian/issues/634
-    - cd ghc/libraries/Cabal/
-    - git remote add quasicomputational https://github.com/quasicomputational/cabal.git
-    - git fetch quasicomputational && git checkout cwd-independent-check
-    - cd ../../../
-
     # Install Alex and Happy
     - set PATH=C:\Users\appveyor\AppData\Roaming\local\bin;%PATH%
     - ghc\hadrian\stack install --install-ghc alex happy > nul

--- a/circle.yml
+++ b/circle.yml
@@ -31,10 +31,6 @@ compile:
     # in CircleCI is a separate process, thus you can't "cd" for the other lines
     - cd ghc/hadrian; git reset --hard HEAD
 
-    # We checkout a patched Cabal.
-    # See: https://github.com/snowleopard/hadrian/issues/634
-    - cd ghc/libraries/Cabal/ && git remote add quasicomputational https://github.com/quasicomputational/cabal.git && git fetch quasicomputational && git checkout cwd-independent-check
-
     - cd ghc; ./boot && PATH=~/.cabal/bin:$PATH ./configure
 
     # XXX: export PATH doesn't work well either, so we use inline env

